### PR TITLE
Validate aggregate window selection, allow recalling windowing UDFs

### PR DESCRIPTION
**Pull request by jacksonrnewhouse**

### CODE REVIEW

File: crates/arroyo-df/src/plan.rs [hunk 1]
```diff
@@ -171,43 +181,138 @@ impl TreeNodeRewriter for AggregateRewriter {
...
+                        if *existing_window != window {
+                            return Err(DataFusionError::Plan(
+                                "window expressions do not match".to_string(),
+                            ));
+                        }
+                        self.fields.insert(projection.schema.field(index).clone());
+                    } else {
+                        // If the input doesn't have an input window, we shouldn't be creating a window.
+                        return Err(DataFusionError::Plan(
+                            "can't call window function in a select statement".to_string(),
```
Comment (mwylde) [inline crates/arroyo-df/src/plan.rs:260]: As a user, what does this mean?

File: crates/arroyo-df/src/plan.rs [hunk 2]
```diff
@@ -171,43 +181,138 @@ impl TreeNodeRewriter for AggregateRewriter {
...
+                            .transpose()
+                    })
+                    .collect::<DFResult<Vec<_>>>()?;
+                self.fields.clear();
+                for (index, window) in window_expressions {
+                    // if there's already a window they should match
+                    if let Some(existing_window) = &self.window {
+                        if *existing_window != window {
+                            return Err(DataFusionError::Plan(
+                                "window expressions do not match".to_string(),
```
Comment (mwylde) [inline crates/arroyo-df/src/plan.rs:253]: can this be explained more clearly?

**Review by mwylde — APPROVED** [review]
A few error messages that I think can be improved, but otherwise looks good!

File: crates/arroyo-df/src/plan.rs (hunk 1 above)
Comment (jacksonrnewhouse) [reply crates/arroyo-df/src/plan.rs]: Gave it another shot. This is the error if you call one of our window UDFs without grouping by it.

Comment (mwylde) [reply crates/arroyo-df/src/plan.rs]: 👍 that's much more clear

**Pull request merged**